source/common/upstream/health_checker_impl.cc

```diff
@@ -113,8 +109,16 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
       codec_client_type_(codecClientType(config.http_health_check().use_http2()
                                              ? envoy::type::HTTP2
                                              : config.http_health_check().codec_client_type())) {
+
+  // The deprecated service_name field was previously being used to compare with the health checked
+  // cluster name using a StartsWith comparison. Since StartsWith is essentially a prefix
+  // comparison, representing the intent by using a StringMatcher prefix is a more natural way.
   if (!config.http_health_check().service_name().empty()) {
```

Comment:
@mattklein123 Were you suggesting to put the comment here? or in the proto definition of `service_name` ?

Reply:
This is fine, thanks.